subsys/net/lib/mqtt_helper/mqtt_helper.c

```diff
@@ -383,7 +384,24 @@ static int broker_init(struct sockaddr_storage *broker,
 	addr = result;
 
 	while (addr != NULL) {
-		if (addr->ai_addrlen == sizeof(struct sockaddr_in)) {
+		/* Check for IPv6 first and use if available. */
+		if (addr->ai_addrlen == sizeof(struct sockaddr_in6)) {
```

Comment:
I will suggest checking `addr->ai_family` instead of comparing lengths and sizes.

```suggestion
		if ((addr->ai_family == AF_INET6) && IS_ENABLED(CONFIG_NET_IPV6)) {
```

Reply:
Ill remove this commit, IPv6 support added in
https://github.com/nrfconnect/sdk-nrf/commit/4f92b8d1f417e5c3b02e912bffa1254a907276cb